go/vt/discovery/tablet_picker_test.go

```diff
@@ -416,6 +441,33 @@ func TestPickError(t *testing.T) {
 	defer cancel()
 	_, err = tp.PickForStreaming(ctx)
 	require.EqualError(t, err, "context has expired")
+	// if local preference is selected, tp cells include's the local cell's alias
+	require.Greater(t, globalTPStats.noTabletFoundError.Counts()["cell_cella.ks.0.replica"], int64(0))
+}
+
+func TestPickError_OnlySpecified(t *testing.T) {
+	te := newPickerTestEnv(t, []string{"cell"})
+
+	tp, err := NewTabletPicker(context.Background(), te.topoServ, te.cells, "cell", te.keyspace, te.shard, "replica", TabletPickerOptions{CellPreference: "OnlySpecified"})
+	require.NoError(t, err)
+	delay := GetTabletPickerRetryDelay()
+	defer func() {
+		SetTabletPickerRetryDelay(delay)
+	}()
+	SetTabletPickerRetryDelay(11 * time.Millisecond)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Millisecond)
+	defer cancel()
+	// no tablets
+	_, err = tp.PickForStreaming(ctx)
+	require.EqualError(t, err, "context has expired")
+	// no tablets of the correct type
+	defer deleteTablet(t, te, addTablet(te, 200, topodatapb.TabletType_RDONLY, "cell", true, true))
+	ctx, cancel = context.WithTimeout(context.Background(), 20*time.Millisecond)
+	defer cancel()
+	_, err = tp.PickForStreaming(ctx)
+	require.EqualError(t, err, "context has expired")
```

Comment:
It's not obvious to me how we're testing OnlySpecified here? Am I missing something?

Reply:
this is showing that if the OnlySpecified option is passed into the tablet picker, but no tablet exists of that specified type, then picker cannot select a tablet